### PR TITLE
[RyuJIT/ARM32] Fix genIsValidDoubleReg assertions occurred with JitStressRegs options

### PR DESCRIPTION
**Pull request by sjsinju**
https://github.com/dotnet/coreclr/issues/13751

Currently, In some cases, It was not used updateAssignedInterval to set the interval.
On ARM architecture, updateAssignedInterval should be used to set the interval for TYP_DOUBLE.

And this PR suggests updateDoubleAssignedInterval, which handles the case that the old Interval is TYP_DOUBLE and the Interval to be allocated is TYP_FLOAT.

### CODE REVIEW

**Comment by hqueue** [comment]
@CarolEidt Please take a look.

File: src/jit/lsra.cpp [hunk 1]
```diff
@@ -5823,6 +5823,10 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
             assert((bestScore & VALUE_AVAILABLE) == 0);
         }
         assignPhysReg(availablePhysRegInterval, currentInterval);
+#ifdef _TARGET_ARM_
+        if (currentInterval->registerType == TYP_DOUBLE && !genIsValidDoubleReg(availablePhysRegInterval->regNum))
```
Comment (CarolEidt) [inline src/jit/lsra.cpp]: I need to understand why we would allocate a register that is not a valid double register.

File: src/jit/lsra.cpp [hunk 2]
```diff
@@ -12821,12 +12863,19 @@ void LinearScan::verifyFinalAllocation()
...
-                            interval->assignedReg             = &(physRegs[regNum]);
-                            interval->isActive                = true;
-                            physRegs[regNum].assignedInterval = interval;
+                            regNumber regNum      = getVarReg(inVarToRegMap, varIndex);
+                            interval              = getIntervalForLocalVar(varIndex);
+                            interval->physReg     = regNum;
+                            interval->assignedReg = &(physRegs[regNum]);
+                            interval->isActive    = true;
+#ifdef _TARGET_ARM_
+                            if (physRegs[regNum].registerType == TYP_DOUBLE)
```
Comment (CarolEidt) [inline src/jit/lsra.cpp]: None of the `RegRecord`s are `TYP_DOUBLE`; they are all set as `TYP_FLOAT`, see https://github.com/dotnet/coreclr/blob/master/src/jit/lsra.h#L222

Comment (sjsinju) [reply src/jit/lsra.cpp]: That's my mistake, I got it. Thank you!

File: src/jit/lsra.cpp (hunk 1 above)
Comment (hqueue) [reply src/jit/lsra.cpp]: I also think so, and it should be an assertion instead.
For example, 
```
assert( currentInterval->registerType != TYP_DOUBLE
           || (currentInterval->registerType == TYP_DOUBLE && genIsValidDoubleReg(availablePhysRegInterval->regNum))
```

Comment (sjsinju) [reply src/jit/lsra.cpp]: OK. I can check it. Thank you.

**Comment by CarolEidt** [comment]
@sjsinju @hqueue I hope you don't mind; I have proposed a different fix.
It is #14573

**Comment by sjsinju** [comment]
@CarolEidt That seems to be better this. Thanks!

**Comment by BruceForstall** [comment]
@sjsinju Looks like you can close this now; #14573 has been merged.

**Pull request closed without merging**